### PR TITLE
Don't install ivadomed just to run pre-commit checks.

### PR DESCRIPTION
**Pull request by kousu**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

It takes 90s to install ivadomed to run a 12s check. And, the longer CI is clogged up waiting for VMs to be allocated, the slower all PRs are. Thus, the time to run pre-commit drastically reduces from ~ 2.45 min to ~ 45 min.

### CODE REVIEW

**Comment by kousu** [comment]
I'm unsure what to do about

https://github.com/ivadomed/ivadomed/blob/85d83421004de8c6af15f532017d02fdb5549a41/setup.py#L40-L43

vs 

https://github.com/ivadomed/ivadomed/blob/85d83421004de8c6af15f532017d02fdb5549a41/.github/workflows/pre-commit.yml#L39

 I guess just keep it in sync manually?

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Install dependencies
         run: |
               python -m pip install --upgrade pip
-              pip install -e .[dev]
+              pip install 'pre-commit>=2.10.1' 'flake8'
```
Comment (kanishk16) [inline .github/workflows/pre-commit.yml:39]: I guess you'd just need pre-commit... IIUC the lint check is initiated in `run_tests.yml`

**Comment by kanishk16** [comment]
> I'm unsure what to do about
> 
> https://github.com/ivadomed/ivadomed/blob/85d83421004de8c6af15f532017d02fdb5549a41/setup.py#L40-L43
> 
> vs
> 
> https://github.com/ivadomed/ivadomed/blob/85d83421004de8c6af15f532017d02fdb5549a41/.github/workflows/pre-commit.yml#L39
> 
> I guess just keep it in sync manually?

Unfortunately yus, even others follow the same convention

**Review by dyt811 — APPROVED** [review]
Nick, you magnificent bastard! Every time I see your short PRs, I learn something cool and realize how much an imposter I am at what I do daily. Great job. Thanks for the awesome initiative and keep innovating. We need more folks like you.

**Pull request merged**